lib/Cake/Controller/Component/Auth/BasicAuthenticate.php

```diff
@@ -82,6 +82,15 @@ public function authenticate(CakeRequest $request, CakeResponse $response) {
  * @return mixed Either false or an array of user information
  */
 	public function getUser(CakeRequest $request) {
+		if(!isset($_SERVER['PHP_AUTH_USER'])) {
+		    if (isset($_SERVER['HTTP_AUTHORIZATION']) && (strlen($_SERVER['HTTP_AUTHORIZATION']) > 0)) {
+		        list($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW']) = explode(':', base64_decode(substr($_SERVER['HTTP_AUTHORIZATION'], 6)));
```

Comment:
What do you suggest as best approach in this case? A simple search for ":" in the string, a try-catch, an extended BasicAuthenticate class specific for FastCGI PHP, something else?